utils/errors/errors.go

```diff
@@ -0,0 +1,55 @@
+package errors
+
+import (
+	"fmt"
+)
+
+// Error interface
+type Error interface {
+	Error() string
+	Cause() string
+	Data() interface{}
+}
+
+// ErrorBundle creates a new response error
+type ErrorBundle struct {
+	cause string
+	data  interface{}
+}
+
+// New creates a new response error
+func New(cause string, data interface{}) error {
+	return &ErrorBundle{
+		cause,
+		data,
+	}
+}
+
+// Data from error origin
+func (err *ErrorBundle) Data() interface{} {
+	return err.data
+}
+
+// Cause returns the associated cause
+func (err *ErrorBundle) Cause() string {
+	return err.cause
+}
+
+// Error turns into an error
+func (err *ErrorBundle) Error() string {
+	return fmt.Sprintf("error: %s", err.cause)
+}
+
+// HTTPResponseState captures the state of the response to be read by lower fns in the stack
+type HTTPResponseState struct {
```

Comment:
I don't think these should be in a shared errors create - they're specific to HTTP clients. What do you think about reorganizing our HTTP clients so they're grouped under a `clients` module? so we have like `utils/clients/cbr` etc

Reply:
yeah, that makes sense